src/relations/mysql_tls.py

```diff
@@ -73,6 +73,11 @@ def _on_tls_relation_joined(self, _) -> None:
 
     def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         """Enable TLS when TLS certificate available."""
+        if not self.charm.unit_peer_data.get("unit-initialized"):
```

Comment:
should we check that the member-state of the unit is explicitly `online`? what if the unit was rebooted another way and is still recovering or offline until the next time update status runs?

Reply:
Yes, you are right. This will be modified